### PR TITLE
Update relay transaction size limit response to look more like the geth response

### LINKED ISSUE #1752: Update Relay transaction size limit response to look more like the geth response.
### Description

Calling the multicaller contract with large sums of data to encode will cause the relay to return a gas limit error response.  Geth returns a transaction size based response:

relay: 
```
ProviderError: [Request ID: 87feb184-eddc-4003-bdb0-406dd6808604] Transaction gas limit '15383176' exceeds block gas limit '15000000'
      at HttpProvider.request (/Users/user/projects/mshakeg/hedera-json-rpc-relay/tools/hardhat-example/node_modules/hardhat/src/internal/core/providers/http.ts:88:21)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async EthersProviderWrapper.send (/Users/user/projects/mshakeg/hedera-json-rpc-relay/tools/hardhat-example/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

Geth:
```
     ProviderError: oversized data: transaction size 480186, limit 131072
      at HttpProvider.request (/Users/user/projects/mshakeg/hedera-json-rpc-relay/tools/hardhat-example/node_modules/hardhat/src/internal/core/providers/http.ts:88:21)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async EthersProviderWrapper.send (/Users/user/projects/mshakeg/hedera-json-rpc-relay/tools/hardhat-example/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```
 
Add a size limit on eth_sendRawTransaction to not be more than 128kb.   [Geth reference](https://github.com/ethereum/go-ethereum/blob/master/core/txpool/legacypool/legacypool.go#L53)
### Steps to reproduce

The following repository has a good reproducer:  [mshakeg](https://github.com/mshakeg/hedera-json-rpc-relay.git)

The test:
`should be able to make 1500 or more multicall processLongInput transactions`

1. Update the hardhat.config.ts to point to the relay.
2. Run the test.
3. Update the hardhat.config.ts to point to Alchemy or Infura.  Sepolia was used to test this.
4. Run the test.

### Additional context

_No response_

### Hedera network

other

### Version

v0.31.3

### Operating system

macOS

### PR DESCRIPTION
**Pull request by konstantinabl**
**Description**:

This PR adds transaction size limit - 128kb in order to conform with geth.
* Adds size check in precheck for sendRawTransaction
* Adds new type of error
* Adds TRANSACTION_SIZE_LIMIT variable to env
* Adds and acceptance test
**Related issue(s)**:

Fixes #1752 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/errors/JsonRpcError.ts [hunk 1]
```diff
@@ -271,4 +271,10 @@ export const predefined = {
       code: -32111,
       message: `Unsupported tracer type - ${type}`,
     }),
+  TRANSACTION_SIZE_TOO_BIG: (size: string) =>
+    new JsonRpcError({
+      name: 'Transaction size too big',
+      code: -32201,
+      message: `Oversized data: transaction size ${size}, transaction limit 131072`,
```
Comment (Ivo-Yankov) [inline packages/relay/src/lib/errors/JsonRpcError.ts:278]: nit: 131072 should not be hardcoded

**Review by Ivo-Yankov — COMMENTED** [review]
LGTM, apart from some nit comments.

**Comment by Ivo-Yankov** [comment]
Add the new env var to the table in `docs/condiguration.md`

File: packages/relay/src/lib/precheck.ts [hunk 2]
```diff
@@ -247,4 +247,16 @@ export class Precheck {
       constants.TX_BASE_COST + constants.TX_DATA_ZERO_COST * zeros + constants.ISTANBUL_TX_DATA_NON_ZERO_COST * nonZeros
     );
   }
+
+  /**
+   * Converts hex string to bytes array
+   * @param hex the hex string you want to convert
+   */
+  hexToBytes(hex: string): Uint8Array {
```
Comment (georgi-l95) [inline packages/relay/src/lib/precheck.ts:255]: nit: add some unit tests for this

File: packages/relay/src/lib/eth.ts [hunk 3]
```diff
@@ -1320,6 +1325,12 @@ export class EthImpl implements Eth {
     let interactingEntity = '';
     let originatingAddress = '';
     try {
+      const transactionToBytes: Uint8Array = this.precheck.hexToBytes(transaction);
```
Comment (georgi-l95) [inline packages/relay/src/lib/eth.ts:1328]: Could also add this in the precheck class in additional method. up to you

**Review by georgi-l95 — COMMENTED** [review]
Overall LG

File: packages/relay/tests/lib/precheck.spec.ts [hunk 4]
```diff
@@ -487,4 +487,13 @@ describe('Precheck', async function () {
       expect(intrinsicGasCost).to.be.greaterThan(constants.TX_BASE_COST);
     });
   });
+
+  describe('hexToBytes', async function () {
+    it('should convert a hex string to bytes', () => {
```
[the remark this answers is not in the document]
Comment (konstantinabl) [reply packages/relay/tests/lib/precheck.spec.ts:492]: I will add for 0x and empty string, however I have added type to the input parameter of the function, which is string so null and undefined cannot be passed

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**